packages/flutter/test/gestures/drag_test.dart

```diff
@@ -81,21 +79,21 @@ void main() {
   });
 
   testGesture('Should report most recent point to onStart by default', (GestureTester tester) {
-    HorizontalDragGestureRecognizer drag = HorizontalDragGestureRecognizer();
-    VerticalDragGestureRecognizer competingDrag = VerticalDragGestureRecognizer()
+    HorizontalDragGestureRecognizer? drag = HorizontalDragGestureRecognizer();
```

Comment:
Yeah, the logic here is a bit convoluted. It looks like they were trying to ensure that these detectors are properly disposed on shutdown with the `addTearDown` calls. This would mean that they are disposed even if an exception was thrown in the middle of the test. However, they also want to manually dispose of these as part of the test. If these are non-nullable, we might get the case where they are disposed twice.

But looking more closely at this, it looks like it didn't really need to dispose of them as part of the test. I will update this to clean it up.